src/controllers/orderController.js

```diff
@@ -5,58 +5,100 @@ const { BadRequestError } = require('../errors');
+    if (!orderItems?.length) {
+      throw new BadRequestError(
+        'Unable to create order because the order items are missing.'
+      );
+    }
+
+    // Create an Order document in the database
+    const order = await Order.create({
+      user: req.user.userId,
+      orderItems,
+      subtotal,
+      tax,
+      total,
+    });
 
-  // Perform necessary checks and validations
-  if (!orderItems?.length) {
-    throw new BadRequestError(
-      'Unable to create order because the order items are missing.'
-    );
-  }
+    // Create a payment intent with Stripe
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount: total * 100, // Stripe requires amount in cents
+      currency: 'usd',
+      description: `Order #${order._id}`,
+      metadata: {
+        userId: req.user.userId,
+        orderId: order._id, // key-values
+        totalQuantity: orderItems.reduce(
+          (total, item) => total + item.quantity,
+          0
+        ), // the total number of items in the order
+        totalAlbums: orderItems.length, // the total number of albums purchased in order
+      },
+    });
 
-  // Create an Order document in the database
-  const order = await Order.create({
-    user: req.user.userId,
-    orderItems,
-    subtotal,
-    tax,
-    total,
-  });
+    // Save the paymentIntent to the Order model
+    order.paymentIntentId = paymentIntent.id;
+    await order.save();
 
-  // Create a payment intent with Stripe
-  const paymentIntent = await stripe.paymentIntents.create({
-    amount: total * 100, // Stripe requires amount in cents
-    currency: 'usd',
-    description: `Order #${order._id}`,
-    metadata: {
-      userId: req.user.userId,
-      orderId: order._id, // key-values
-      totalQuantity: orderItems.reduce(
-        (total, item) => total + item.quantity,
-        0
-      ), // the total number of items in the order
-      totalAlbums: orderItems.length, // the total number of albums purchased in order
-    },
-  });
+    // Create PurchasedAlbum entries for each album in orderItems
+    for (const orderItem of orderItems) {
+      await PurchasedAlbum.create({
+        album: orderItem.album, // album ID from orderItem
+        user: req.user.userId, // user ID from the request
+      });
+    }
 
-  // Save the paymentIntent to the Order model
-  order.paymentIntentId = paymentIntent.id;
-  await order.save();
 
-  // Create PurchasedAlbum entries for each album in orderItems
-  for (const orderItem of orderItems) {
-    await PurchasedAlbum.create({
-      album: orderItem.album, // album ID from orderItem
-      user: req.user.userId, // user ID from the request
-    });
-  }
 
-  // Send the payment intent client secret and order information to the client
-  res
-    .status(StatusCodes.CREATED)
-    .json({ clientSecret: paymentIntent.client_secret, order });
+    // Start a database transaction
+    await db.startTransaction();
```

Comment:
Rather than creating a separate database connection and starting a transaction that way, we can use the existing connection(s) via our model: [https://mongoosejs.com/docs/transactions.html#:~:text=In%20practice%2C%20you,transaction%20error.](https://mongoosejs.com/docs/transactions.html#:~:text=In%20practice%2C%20you,transaction%20error.)